imports/plugins/included/product-detail-simple/client/templates/productReview.js

```diff
@@ -0,0 +1,45 @@
+import { Template } from "meteor/templating";
+import { Meteor } from "meteor/meteor";
+import { Reaction } from "/client/api";
+import { Reviews } from "/lib/collections";
+import { Products } from "/lib/collections";
+import "./productReview.html";
+
+const review = {};
+Template.submitReview.events({
+  "click .stars": () => {
+    const rating = $("#rating").data("userrating");
+    review.rating = rating;
+  },
+  "click #submit": () => {
+    review.comment = document.getElementById("comment").value;
+    if (review.comment === "") {
+      Alerts.toast("Comment cannot be empty", "error");
+      return false;
+    }
+    this.productId = () => Reaction.Router.getParam("handle");
+    review.productId = Products.findOne(this.productId)._id;
+    try {
+      review.username = Meteor.user().username || Meteor.user().emails[0].address;
+      review.dateCreated = new Date;
+      Meteor.call("insert/review", review, function (error) {
+        if (error) {
+          return error;
+        }
+      });
+      document.getElementById("comment").value = "";
+    } catch (error) {
+      Alerts.toast("You need to sign in to post a review", "error");
+    }
+  }
+});
+
+Template.displayReviews.helpers({
+  reviews: () => {
+    this.productId = () => Reaction.Router.getParam("handle");
```

Comment:
why is this a function?

Reply:
A function is required here to get the actual reference to the clicked product being reviewed. Using a regular variable will get the id of the product's ancestor (if the product has variants) which is not always the product being reviewed